### ISSUE TITLE
no-unused-vars reports on class constructors where class methods are used

### ISSUE THREAD

**Issue by himynameisdave** [eslint/eslint#11533]
**Tell us about your environment**

* **ESLint Version:** `5.15.3` (latest as of writing)
* **Node Version:** `v10.14.1`
* **npm Version:** `6.8.0`

**What parser (default, Babel-ESLint, etc.) are you using?**

- [`babel-eslint`](https://github.com/babel/babel-eslint)

**Please show your full configuration:**

- [`eslint-config-7geese`](https://github.com/7Geese/eslint-config-7geese)

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
export const dismissAllNotifications = (notifications) => {
    return (dispatch) => {
        const notificationsLength = notifications.length;
        dispatch(removeAllNotifications(notifications));
        for (let i = 0; i < notificationsLength; i++) {
            const notification = notifications[i];
            const notificationModel = new Notification({ id: notification.id });
            notificationMode.destroy();
        }
        return 'dismissed';
    };
};
```

```bash
eslint src/ --ext .jsx,.js --quiet --report-unused-disable-directives
```

![image](https://user-images.githubusercontent.com/4298089/54717264-03b37600-4b15-11e9-8d94-dec3e1de8d53.png)

**What did you expect to happen?**

I wouldn't expect `no-unused-vars`/`no-undef` to be

**What actually happened? Please include the actual, raw output from ESLint.**

```
.../redux/home-actions.js
  207:19  error  'notificationModel' is assigned a value but never used  no-unused-vars
  208:13  error  'notificationMode' is not defined                       no-undef
```

**Are you willing to submit a pull request to fix this bug?**

Absolutely, I would just need to be pointed in the right direction 👍

**Comment by not-an-aardvark** [comment]
Hi, thanks for creating an issue.

Based on the code you've provided, I'm confused about why the errors are unexpected. Is there another place in the code where `notificationMode` is used/where `notificationModel` is defined? From the code you've provided, the linting errors seem correct to me , but I may be missing some context.

**Comment by not-an-aardvark** [comment]
Also, just to make sure: are `notificationModel` and `notificationMode` supposed to be two separate things? If one of them is a typo, then fixing the typo would also remove both linting errors.

**Comment by himynameisdave** [comment]
👍 Yep I think this is just a typo/bad var scoping in our codebase. Sorry for the issue, should have read it harder! 

![](https://media.giphy.com/media/4bpK2k0Yru5Us/giphy.gif)
